src/components/dock/dock-button/dock-button.spec.tsx

```diff
@@ -0,0 +1,35 @@
+import { h } from '@stencil/core';
+import { newSpecPage, SpecPage } from '@stencil/core/testing';
+import { DockItem } from '../dock.types';
+import { DockButton } from './dock-button';
+
+let page: SpecPage;
+let item: DockItem;
+let eventSpy: jest.Mock;
+
+beforeEach(async () => {
+    item = {
+        id: 'tables',
+        label: 'Tables',
+        icon: 'insert_table',
+        dockMenu: { componentName: 'my-custom-menu' },
+    };
+    page = await newSpecPage({
+        components: [DockButton],
+        template: () => <limel-dock-button item={item} />,
+    });
+    eventSpy = jest.fn();
+    page.body.addEventListener('dockMenuOpened', eventSpy);
+    await page.waitForChanges();
+});
+
+test('emits an event with the menu item when the menu item in the dock is clicked', async () => {
+    const button = (await page.root.shadowRoot.querySelector(
```

Comment:
```suggestion
    const button = (await page.root.querySelector(
```

Reply:
Yea that was implemented when the shadow DOM was true.